bundles/plugins/org.bonitasoft.studio.dependencies/src/org/bonitasoft/studio/dependencies/handler/ImportLibsRunnable.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (C) 2015 Bonitasoft S.A.
+ * Bonitasoft, 32 rue Gustave Eiffel - 38000 Grenoble
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2.0 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.bonitasoft.studio.dependencies.handler;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.zip.ZipEntry;
+
+import org.bonitasoft.studio.common.ZipInputStreamIFileFriendly;
+import org.bonitasoft.studio.common.log.BonitaStudioLog;
+import org.bonitasoft.studio.common.repository.Messages;
+import org.bonitasoft.studio.dependencies.repository.DependencyRepositoryStore;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jface.operation.IRunnableWithProgress;
+import org.eclipse.swt.widgets.Display;
+
+public class ImportLibsRunnable implements IRunnableWithProgress {
+
+    private static final String ZIP_EXTENSION = ".zip";
+    private static final String JAR_EXTENSION = ".jar";
+    private final DependencyRepositoryStore libStore;
+    private final String[] jarAndZips;
+    private final String filterPath;
+
+    public ImportLibsRunnable(final DependencyRepositoryStore libStore, final String[] jarAndZips, final String filterPath) {
```

Comment:
We use this terminology everywhere else for IRunnableWithProgress...
It's inspired from Eclipse sources. I found more expressive than Runnable.